File: platforms/ios/Dezel/Dezel/View/Animation/TransitionListener.swift
/**
 * @protocol Transitionable
 * @since 0.2.0
 */
public protocol TransitionListener: AnyObject {

	/**
	 * @method shouldBeginTransitionAnimation
	 * @since 0.2.0
	 */
	func shouldBeginTransitionAnimation(animation: CABasicAnimation, for property: String, of layer: CALayer) -> Bool

	/**
	 * @method willBeginTransitionAnimation
	 * @since 0.2.0
	 */
	func willBeginTransitionAnimation(animation: CABasicAnimation, for property: String, of layer: CALayer)

	/**
	 * @method didCommitTransition
	 * @since 0.6.0
	 */
	func didCommitTransition()

	/**
	 * @method didFinishTransition
	 * @since 0.2.0
	 */
	func didFinishTransition()
}

File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/view/animation/TransitionListener.kt
package ca.logaritm.dezel.view.animation

import android.animation.ValueAnimator

/**
 * The interface for transitionable elements.
 * @interface TransitionListener
 * @since 0.2.0
 */
public interface TransitionListener {

	/**
	 * Called when a transition animation is created for the specified property.
	 * @method onBeforeTransitionBegin
	 * @since 0.2.0
	 */
	fun onBeforeTransitionBegin(animator: ValueAnimator, property: String)

	/**
	 * Called when the transition animation begins.
	 * @method onTransitionBegin
	 * @since 0.2.0
	 */
	fun onTransitionBegin()

	/**
	 * Called when the transition animation finishes.
	 * @method onTransitionFinish
	 * @since 0.2.0
	 */
	fun onTransitionFinish()
}